### PR TITLE
Rename 'src/bootstrap/native.rs' to llvm.rs

### LINKED ISSUE #108240: Rename `src/bootstrap/native.rs` to `llvm.rs`
The comment at the top of the file says https://github.com/rust-lang/rust/blob/ae5473c9699eaf1998c414e0a8884383bb9ddcc3/src/bootstrap/native.rs#L1 so presumably the logic is that "native" includes C/C++ code that isn't LLVM. But the only C code that's not LLVM is https://github.com/rust-lang/rust/blob/ae5473c9699eaf1998c414e0a8884383bb9ddcc3/src/bootstrap/native.rs#L899 I think we should move `TestHelpers` to `test.rs` and rename `native.rs` to `llvm.rs` to make it easier to find for new contributors.

cc @compiler-errors, I think you asked about where to find this code recently.

### PR DESCRIPTION
**Pull request by rohaquinlop**
Fixed #108240

Renamed 'native.rs' to 'llvm.rs', also moved `TestHelpers` to `test.rs`. Replaced all the `native.rs` occurrences at `src/bootstrap` files to `llvm.rs`

### CODE REVIEW

**Comment by albertlarsan68** [comment]
Please [rebase](https://rustc-dev-guide.rust-lang.org/git.html#rebasing) on top of new `master` instead of merging.
@rustbot author

**Comment by albertlarsan68** [comment]
Not quite, try again! :)

**Comment by rohaquinlop** [comment]
@albertlarsan68 Should I rebase my fork master with the branch I created with my changes? I don't get it, it's my first time 😅

**Comment by albertlarsan68** [comment]
I would run `git rebase upstream/master`, with the `upstream` remote being an up-to-date remote of the rust-lang/rust repo, and then run `git push --force-with-lease`.

**Comment by albertlarsan68** [comment]
I rebased for you.

File: src/ci/stage-build.py [hunk 1]
```diff
@@ -175,8 +175,8 @@ def rustc_stage_2(self) -> Path:
         return super().rustc_stage_2().with_suffix(".exe")
 
     def build_rustc_perf(self):
-        # rustc-perf version from 2023-03-15
-        perf_commit = "9dfaa35193154b690922347ee1141a06ec87a199"
+        # rustc-perf version from 2022-07-22
+        perf_commit = "3c253134664fdcba862c539d37f0de18557a9a4c"
```
[the remark this answers is not in the document]
Comment (albertlarsan68) [reply src/ci/stage-build.py:179]: The tree from the old HEAD and the new is the same, so the PR was already broken.

[the remark this answers is not in the document]
Comment (rohaquinlop) [reply src/ci/stage-build.py:179]: What should I do?

[the remark this answers is not in the document]
Comment (albertlarsan68) [reply src/ci/stage-build.py:179]: Amend the commit with this change reverted, and force-push your branch.

[the remark this answers is not in the document]
Comment (rohaquinlop) [reply src/ci/stage-build.py]: Done.

**Comment by rohaquinlop** [comment]
@albertlarsan68 Guess everything is Ok now, I'll be pending if anything goes wrong (Hope this isn't the case 😅).

File: src/bootstrap/llvm.rs [hunk 2]
```diff
@@ -568,8 +568,6 @@ fn configure_cmake(
             cfg.define("CMAKE_SYSTEM_NAME", "Haiku");
         } else if target.contains("solaris") || target.contains("illumos") {
             cfg.define("CMAKE_SYSTEM_NAME", "SunOS");
-        } else if target.contains("linux") {
-            cfg.define("CMAKE_SYSTEM_NAME", "Linux");
```
Comment (klensy) [inline src/bootstrap/llvm.rs:572]: There still few places where code removed, is it intended?

Comment (rohaquinlop) [reply src/bootstrap/llvm.rs:572]: No, it isn't intended. I just ran `git rebase` but the native content wasn't modified.

Comment (rohaquinlop) [reply src/bootstrap/llvm.rs]: Reverted the changes that I didn't realized

**Comment by klensy** [comment]
Looking at diff +101/-101, looks correct.

**Comment by rohaquinlop** [comment]
> Looking at diff +101/-101, looks correct.

Great!

**Comment by Mark-Simulacrum** [comment]
Thanks! @bors r+

**Pull request merged**